bin/specextract

```diff
@@ -79,381 +81,449 @@ v4 = make_verbose_level(toolname, 4)
+                mkrmf.clobber = clobber
+                mkrmf.verbose = str(verbose)
+
+                mkrmf()
 
-        mkrmf.infile = "CALDB"
-        mkrmf.outfile = rmffile
-        mkrmf.logfile = ""
-        mkrmf.weights = weightfile
-        mkrmf.axis1 = f"energy={ebin}"
-        mkrmf.axis2 = rmfbin
-        mkrmf.clobber = clobber
-        mkrmf.verbose = str(verbose)
+            else:
+                channel = rmfbin.split("=")
+                channel.reverse()
+
+                mkacisrmf.punlearn()
+
+                # force CALDB querry to match 'ccd_id' parameter, otherwise the default
+                # behavior is to use the CCD_ID header keyword in the 'obsfile'
+                mkacisrmf.infile = f"CALDB(CCD_ID={ccd_id})"
+                mkacisrmf.outfile = rmffile
+                mkacisrmf.energy = ebin
+                mkacisrmf.channel = channel[0]
+                mkacisrmf.chantype = ptype.upper()
+                mkacisrmf.wmap = "none"
+                mkacisrmf.gain = "CALDB"
+                mkacisrmf.obsfile = evt_filename
+                mkacisrmf.ccd_id = ccd_id
+                mkacisrmf.chipx = chipx
+                mkacisrmf.chipy = chipy
+                mkacisrmf.clobber = clobber
+                mkacisrmf.verbose = str(verbose)
+
+                mkacisrmf()
 
-        mkrmf()
-        
         return rmffile
 
-    elif "mkacisrmf" == rmftool:
-
-        channel = rmfbin.split("=")
-        channel.reverse()
-
-        mkacisrmf.punlearn()
-
-        mkacisrmf.infile = "CALDB"
-        mkacisrmf.outfile = rmffile
-        mkacisrmf.energy = ebin
-        mkacisrmf.channel = channel[0]
-        mkacisrmf.chantype = ptype.upper()
-        mkacisrmf.gain = "CALDB"
-        mkacisrmf.clobber = clobber
-        mkacisrmf.verbose = str(verbose)
-        mkacisrmf.ccd_id = "0" # hopefully, this par will
-                               # be ignored since it can't
-                               # be set to "".
-        if wmap_clip:
-            mkacisrmf.wmap = f"{wmap_sky2tdet}[wmap]"
-        else:
-            mkacisrmf.wmap = f"{specfile}[WMAP]" # dmextract WMAP is faster than sky2tdet WMAP
-
-        mkacisrmf()
-        
-        return rmffile
-        
-    else:
+    except (OSError,IOError):
         raise IOError(f"Failed to find an appropriate tool to generate a RMF for {specfile}")
 
+    
 
-##########################################################################
-# Usage:
-#   build_rmf_ps(rmftool, ptype, full_outroot, ebin,
-#              rmfbin, clobber, verbose, specfile, weightfile, ccd_id, chipx, chipy)
-#
-#
-# Aim:
-#   Run either mkrmf or mkacisrmf depending on input conditions.
-#   For mkrmf:  input FEF file and no weight file and return RMF name.
-#   For mkacisrmf:  input CALDB and WMAP and return RMF name.
-#
-##########################################################################
+def build_rmf_ext(rmftool, ptype, full_outroot, ebin, rmfbin, clobber, verbose,
```

Comment:
So, without manually going line by line, it looks like this and `build_rmf_ps` could just call a `builf_rmf_basic` routine where you send in the arguments. This may be a bit awkward (passing through too many arguments) so maybe it wouldn't be helpful. Maybe there should be two low-level routines (one for mkrmf and one for mkacisrmf) that are called by `build_rmf_ps` and `build_rmf_ext`.